libraries/AP_HAL_ChibiOS/Semaphores.h

```diff
@@ -37,3 +37,21 @@ class ChibiOS::Semaphore : public AP_HAL::Semaphore {
     // we declare the lock as a uint32_t array, and cast inside the cpp file
     uint32_t _lock[5];
```

Comment:
I think it would be clearer to have something like "#define SIZEOF_BINARY_SEMAPHORE 5", also the uint32_t is presumably for alignment, wouldn't void* be more correct?

Reply:
we don't actually know the size, as it can vary with compile options (debug, assert etc). We use a static assert in the cpp file to ensure it is big enough

Reply:
look further up in Semaphores.h for the exact same pattern for a mutex